cobra/core/Model.py

```diff
@@ -118,6 +117,45 @@ def description(self, value):
         self.name = value
         warn("description deprecated", DeprecationWarning)
 
+    @property
+    def medium(self):
+        return {rxn.id: rxn.bounds for rxn in
+                self.reactions.query(lambda x: x, 'boundary')}
+
+    @medium.setter
+    def medium(self, medium):
+        """Get or set the constraints on the model exchanges.
+
+        `model.medium` returns a dictionary of the bounds for each of the
+        boundary reactions, in the form of `{rxn_id: (l_bound, u_bound)}`
+
+        Parameters
+        ----------
+        medium: dictionary-like
+            The medium to initialize. medium should be a dictionary defining
+            `{rxn_id: bounds}` pairs.
+
+        """
+
+        # Check the symmetric difference to see if reactions are missing
+        boundary_rxn_keys = set([rxn.id for rxn in self.reactions.query(
+                        lambda x: x.boundary)])
+        medium_keys = set(medium.keys())
+        missing_in_medium = boundary_rxn_keys - medium_keys
+        missing_in_model = medium_keys - boundary_rxn_keys
+
+        for rxnid in missing_in_medium:
+            warn("{} is missing from the medium".format(rxnid))
```

Comment:
So there's really two issues here it seems,

1) I don't think we should be zero'ing out reaction bounds which arn't explicitly included in the passed medium. Largely as a result of my point above, which is that if model.medium returns export reactions a user isn't expecting, then it would zero those. So the warning on missing reactions was essentially because the desired behavior in this instance isn't obvious.

2) There's some back-and-forth on whether having extraneous reactions in the passed media should cause a warning or error. On one hand, it could just be a misspelling of the desired reaction (should let the user know), on the other, the media dict is being applied to many models, some of which have extra exchange reactions (see @mmundy42's above). Hence the warning here.

I would obviously prefer an approach which didn't use the warnings. But that would definitely require a firm consensus on what model.medium is for 😄 .